### PR TITLE
Upgrade to rdflib 0.15.0, update PATCH tests

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Addresses issue #491.
Fixes PATCH- and Glob-related failed tests when upgrading to rdflib 0.15.0

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
cc @timbl / @dan-f for review.

File: test/patch-2.js [hunk 1]
```diff
@@ -109,10 +133,7 @@ describe('PATCH', function () {
         .end(function (err, res, body) {
           assert.equal(
             read('sampleContainer/prefixSparql.ttl'),
-            '@prefix schema: <http://schema.org/>.\n' +
-            '@prefix profile: <http://ogp.me/ns/profile#>.\n' +
-            '\n' +
-            '   <#> profile:first_name "Timothy"; a schema:Person .\n')
+            '@prefix : </prefixSparql.ttl#>.\n@prefix schema: <http://schema.org/>.\n@prefix pro: <http://ogp.me/ns/profile#>.\n\n: a schema:Person; pro:first_name "Timothy".\n\n')
```
Comment (dan-f) [inline test/patch-2.js]: A bit confused about this change given the one above. Let's be consistent with formatting of data. I think the previous change is much more readable.

File: lib/handlers/get.js [hunk 2]
```diff
@@ -176,7 +176,7 @@ function globHandler (req, res, next) {
       var data = $rdf.serialize(
         undefined,
         globGraph,
-        null,
+        filename,
         'text/turtle')
```
Comment (dmitrizagidulin) [inline lib/handlers/get.js]: From what I understand, glob functionality is only spec'd/implemented for Turtle.

File: test/patch-2.js (hunk 1 above)
Comment (dmitrizagidulin) [inline test/patch-2.js]: No prob, will change it to match.

File: lib/handlers/get.js (hunk 2 above)
Comment (dan-f) [inline lib/handlers/get.js]: From what I understand, the notion of "globbing all the graphs in an LDP container into a single graph" is agnostic to a serialization format.

File: lib/handlers/get.js [hunk 3]
```diff
@@ -176,7 +176,7 @@ function globHandler (req, res, next) {
       var data = $rdf.serialize(
         undefined,
         globGraph,
-        null,
+        filename,
```
Comment (dmitrizagidulin) [inline lib/handlers/get.js]: An asterisk is a valid URI character. But whether that's the preferred semantics, as far as the base uri when using file patterns, is a tough call. We'll need to discuss it with @timbl etc.

File: lib/handlers/get.js (hunk 2 above)
Comment (dmitrizagidulin) [inline lib/handlers/get.js]: We can open an issue to add support for multiple content types for globbing, for solid-server. This PR is just intended to upgrade the server to the latest rdflib version.

File: lib/handlers/get.js (hunk 3 above)
Comment (dan-f) [inline lib/handlers/get.js]: My point was not regarding the asterisk. It was regarding the lack of a URI scheme

Comment (dmitrizagidulin) [inline lib/handlers/get.js]: Ohh I see. Thanks! will fix

File: test/patch-2.js (hunk 1 above)
Comment (dmitrizagidulin) [inline test/patch-2.js]: Changed the expected value formatting to be multi-line. Thanks!

File: lib/handlers/get.js (hunk 2 above)
Comment (dmitrizagidulin) [inline lib/handlers/get.js]: I opened issue #505 to add support for content negotiation with globbing.

File: lib/handlers/get.js (hunk 3 above)
Comment (dmitrizagidulin) [inline lib/handlers/get.js]: Fixed the base serialization uri to be the request uri.

**Comment by dmitrizagidulin** [comment]
> I fail to see how this "fixes Glob-related failed tests". I found several "glob-related tests", but none of them changed in this PR.

I just meant - this PR passes the failing tests that are introduced by upgrading to latest rdflib.

> The first glob-related tests are actually skipped. I tested, on master, un-skipping that describe block, and all of the glob tests failed. This is before the rdflib upgrade.

Yeah, that's weird, I'm not sure what the story was with those. I added an issue, #505, to un-skip and implement those tests.

> The second "glob test" doesn't test globbing. It simply tests for a 200 response code and content-type text/turtle.

Yeah, also odd. I expanded that test to actually check for result contents.

**Comment by melvincarvalho** [comment]
A gentle bump on this.  

Hope it's OK, that I added the blocker label, as this fix will enable me to log in to node solid server, and therefore, work more effectively.

File: test/resources/sampleContainer/example1.ttl [hunk 4]
```diff
@@ -2,6 +2,8 @@
 @prefix dc: <http://purl.org/dc/elements/1.1/> .
 @prefix ex: <http://example.org/stuff/1.0/> .
 
+</sampleContainer/example*> a :glob.
```
Comment (dmitrizagidulin) [inline test/resources/sampleContainer/example1.ttl]: Added to address @dan-f's comment; tests the `requestUri` being used as a serialization base for globs. (per @timbl's instructions)

Comment (dan-f) [inline test/resources/sampleContainer/example1.ttl]: Uh... I would expect this triple to be *dynamically* added by the glob code, not *statically* added to the `sampleContainer/example1.ttl` graph.

Comment (dmitrizagidulin) [inline test/resources/sampleContainer/example1.ttl]: This isn't an official designation or anything, just a triple @timbl recommended to test the serialization base in the output.

Comment (dan-f) [inline test/resources/sampleContainer/example1.ttl]: I find it a really confusing example. Why not just add <#a> dc:title "foo" . to example1.ttl and then expect to see <example1.ttl#a> dc:title "foo" . in the glob?

I don't want to hold this up any longer

Comment (dmitrizagidulin) [inline test/resources/sampleContainer/example1.ttl]: The example tests the serialization base being the glob request uri. The one you mention, `<#a> dc:title "foo" .` tests the parsing base (relative to example1.ttl). I'll add this line also.

Comment (dan-f) [inline test/resources/sampleContainer/example1.ttl]: Huh. Sounds like I am missing some context from your discussion. I think you should merge if you and @timbl are happy with the test.

Comment (deiu) [inline test/resources/sampleContainer/example1.ttl]: FYI, this is not how you test glob. `</sampleContainer/example*> a :glob.` being in a document makes no sense to me.

Comment (dmitrizagidulin) [inline test/resources/sampleContainer/example1.ttl]: @deiu would something like `</sampleContainer/example*> :test "testing serialization base" .` make more sense, instead? 

I don't have a stake in this argument; was just adding the statement at @timbl's suggestion.

Comment (deiu) [inline test/resources/sampleContainer/example1.ttl]: First, you don't have to keep this PR open on my account. Please merge if you think it's the right thing to do. I'm only pointing out that having a triple with the subject `/example*` in that  document makes **absolutely no sense**. The `*` character must be removed by the server before it starts to glob resources together, it should never be treated as a regular resource (which is what the test currently does).

Comment (dmitrizagidulin) [inline test/resources/sampleContainer/example1.ttl]: @deiu I've replaced the `<example*> a :glob` statement with the example you & @dan-f suggested.

**Pull request merged**